sourcecred/sourcecred
# Add Payout Address Text Box To Identities UI

This text box allows the entry and modification of payout addresses for the
configured `integrationCurrency` in the instance's config/currency.json.
This narrow scoping was decided upon to minimize the development lift
and complexity for the user.

This input box validates, and verbosely warns the user when an address
is not valid. Currently, it only validates EVM addresses. Valid
addresses can be submitted to the ledger by clicking the Update Name and
address button. If either the name or payout address is unchanged, the
unchanged value is not submitted to the ledger. This is also prevented
on the backend.

As a convenience, entering a valid address, either by selecting an
existing identity that possesses an active payout address or manually
entering one, will cause an etherscan hyperlink to render below the text
box.

![Peek 2021-10-20 15-46](https://user-images.githubusercontent.com/17910833/138169957-fcebbec4-633f-45cd-889f-410b4599c620.gif)


Test Plan
1. `yarn start`
2. browse to http://localhost:8080/#/admin
3. click on the dandelion identity and observe the existing address and
   etherscan link
4. delete the last character of the address and observe the etherscan
   link disappear and the validation notify the user that the address is
   invalid. The update button is also no longer active
5. click on beanow and paste in "0xBFc7CAE0Fad9B346270Ae8fde24827D2D779eF07"
6. Observe the etherscan link appear
7. Click the update button and save the ledger to disk
8. Observe the new Payout address configured in the ledger
